pkg/admission/apiexportendpointslice/apiexportendpointslice_admission.go

```diff
@@ -0,0 +1,201 @@
+			return fmt.Errorf("failed to convert unstructured to APIExportEndpointSlice: %w", err)
+		}
+	}
+
+	switch {
+	case a.GetOperation() == admission.Create,
+		// this may be redundant as APIExportEndpointSlice.Spec.APIExport is immutable
+		a.GetOperation() == admission.Update && !reflect.DeepEqual(slice.Spec.APIExport, oldSlice.Spec.APIExport):
+		// unified forbidden error that does not leak workspace existence
+		// only considering "create" as the APIExport reference is immutable
+		forbidden := admission.NewForbidden(a, fmt.Errorf("unable to create APIExportEndpointSlice: no permission to bind to export %s",
+			logicalcluster.NewPath(slice.Spec.APIExport.Path).Join(slice.Spec.APIExport.Name).String()))
+
+		// get cluster name of export
+		var exportClusterName logicalcluster.Name
+		if slice.Spec.APIExport.Path == "" {
+			exportClusterName = clusterName
+		} else {
+			path := logicalcluster.NewPath(slice.Spec.APIExport.Path)
+			export, err := o.getAPIExport(path, slice.Spec.APIExport.Name)
+			if err != nil {
+				return forbidden
+			}
+			exportClusterName = logicalcluster.From(export)
+		}
+
+		// Access check
+		if err := o.checkAPIExportAccess(ctx, a.GetUserInfo(), exportClusterName, slice.Spec.APIExport.Name); err != nil {
+			return forbidden
+		}
+	}
+
+	return nil
+}
+
+func (o *apiExportEndpointSliceAdmission) checkAPIExportAccess(ctx context.Context, user user.Info, apiExportClusterName logicalcluster.Name, apiExportName string) error {
+	logger := klog.FromContext(ctx)
+	authz, err := o.createAuthorizer(apiExportClusterName, o.deepSARClient)
+	if err != nil {
+		// Logging a more specific error for the operator
+		logger.Error(err, "error creating authorizer from delegating authorizer config")
+		// Returning a less specific error to the end user
+		return errors.New("unable to authorize request")
+	}
+
+	return authorization.CheckAPIExportAccess(ctx, user, apiExportName, authz)
+}
+
+// ValidateInitialization ensures the required injected fields are set.
+func (o *apiExportEndpointSliceAdmission) ValidateInitialization() error {
+	if o.deepSARClient == nil {
+		return fmt.Errorf(PluginName + " plugin needs a Kubernetes ClusterInterface")
+	}
+	if o.apiExportLister == nil {
+		return fmt.Errorf(PluginName + " plugin needs an APIExport lister")
+	}
+	return nil
+}
+
+// SetDeepSARClient is an admission plugin initializer function that injects a client capable of deep SAR requests into
+// this admission plugin.
+func (o *apiExportEndpointSliceAdmission) SetDeepSARClient(client kcpkubernetesclientset.ClusterInterface) {
+	o.deepSARClient = client
+}
+
+func (o *apiExportEndpointSliceAdmission) SetKcpInformers(informers kcpinformers.SharedInformerFactory) {
+	apiExportsReady := informers.Apis().V1alpha1().APIExports().Informer().HasSynced
+	o.SetReadyFunc(func() bool {
+		return apiExportsReady()
+	})
+	o.apiExportLister = informers.Apis().V1alpha1().APIExports().Lister()
```

Comment:
where is this field used?

Reply:
Good point. I am removing it. It is also in the apibinding admission and I could not see any use of it there either. It could be fixed there in a subsequent PR